<launch>

  <arg name="interface" default="can0" />
  <arg name="receiver_interval_sec" default="0.01" />
  <arg name="sender_timeout_sec" default="0.01" />
  <arg name="enable_can_fd" default="false" />

  <include file="$(find-pkg-share ros2_socketcan)/launch/socket_can_receiver.launch.py">
    <arg name="interface" value="$(var interface)" />
    <arg name="interval_sec" value="$(var receiver_interval_sec)" />
    <arg name="enable_can_fd" value="$(var enable_can_fd)" />
  </include>

  <include file="$(find-pkg-share ros2_socketcan)/launch/socket_can_sender.launch.py">
    <arg name="interface" value="$(var interface)" />
    <arg name="timeout_sec" value="$(var sender_timeout_sec)" />
    <arg name="enable_can_fd" value="$(var enable_can_fd)" />
  </include>
  
  <node pkg="canedudev_interface" exec="canedudev_interface_receiver">
    <remap from="/input/can_rx" to="/from_can_bus"/>
  </node>

  <node pkg="canedudev_interface" exec="canedudev_interface_sender" >
    <remap from="/output/can_tx" to="/to_can_bus"/>
    <param name="loop_rate" value="100.0" />
  </node>

</launch>